tests/kontonummer.js

```diff
@@ -0,0 +1,180 @@
+// external
+const chai = require('chai');
+
+// internal
+const kontonummer = require('../kontonummer');
+
+describe('Kontonummer suite', () => {
+  describe('fillZeros()', () => {
+    const mockBank = {
+      name    : 'Swedbank',
+      regex   : /^(8[0-9]{4})(0*[0-9]+)$/,
+      range   : '8000-8999',
+      modulo  : 10,
+      lengths : {
+        clearing : 5,
+        account  : 10,
+        control  : 10
+      },
+      zerofill: true
+    };
+    const fakeAccountNumbers = [
+      '8123412345',
+      '899991',
+      '854321234567890'
+    ];
+    const expectations = [
+      '812340000012345',
+      '899990000000001',
+      '854321234567890'
+    ];
+
+    it('should correctly prepend zeros to account number(s)', () => {
+      fakeAccountNumbers.forEach((accountNumber, i) => {
+        const result = kontonummer._fillZeros(accountNumber, mockBank);
+
+        chai.assert.strictEqual(expectations[i], result);
+      });
+    });
+  });
+
+  describe('getCtrlNum()', () => {
+    it('should return a 10/11 digit control number', () => {
```

Comment:
Sorry if I was unclear. I didn't mean that both asserts need to be in a single test. I think it's good to only have one assert per test to easily see what is wrong if it breaks. However, my point was that by having multiple asserts testing the same function that only uses `substr` internally, basically means you are testing the `substr` method rather than the specifics of _how_ the function is using `substr`.

What is unique about this function, is what part of the string it takes out (X characters from the end). The first assert checks that by ensuring that the first number `1` is not included in the expected return value of `_getCtrlNum()`. However, checking if it picks 11 of 11 characters or 10 of 11 characters is kind of testing the same thing.

If you want to check edge cases for regression testing in case we desire to change the internal implementation, such as because of the latest recommendation in the [9th edition of ECMA-262](https://www.ecma-international.org/ecma-262/9.0/index.html#sec-string.prototype.substr) of not using [`substr`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr) in new code ([`slice`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) could replace it), maybe checking the result of picking 12 characters of 11 would be a more useful test? That could potentially return different things depending on the internal implementation and could happen for real, because of the changes Marcin made in a04b62af27e2df201279609b5a541c986010fa7b that mean the wrong digits may be included in the mod10/11 check, which may produce unexpected results. For example: If a bank account number needs to be 11 characters, the user enters 10 missing the last digit, the mod10 check may still pass, but on the wrong digits.